charts/nri-metadata-injection/templates/_helpers.tpl

```diff
@@ -1,91 +1,66 @@
 {{/* vim: set filetype=mustache: */}}
+
 {{/*
-Expand the name of the chart.
+Renders a value that contains template.
+Usage:
+{{ include "tplvalues.render" ( dict "value" .Values.path.to.the.Value "context" $) }}
 */}}
-{{- define "nri-metadata-injection.name" -}}
-{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- define "tplvalues.render" -}}
+    {{- if typeIs "string" .value }}
+        {{- tpl .value .context }}
+    {{- else }}
+        {{- tpl (.value | toYaml) .context }}
+    {{- end }}
 {{- end -}}
 
-{{/*
-Create a default fully qualified app name.
-We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
-If release name contains chart name it will be used as a full name.
-*/}}
-{{- define "nri-metadata-injection.fullname" -}}
-{{- if .Values.fullnameOverride -}}
-{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
-{{- else -}}
-{{- $name := default .Chart.Name .Values.nameOverride -}}
-{{- if contains $name .Release.Name -}}
-{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
-{{- else -}}
-{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- /*
+Naming helpers
+*/ -}}
+
+{{- define "nri-metadata-injection.name.admission" -}}
+{{ include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" (include "newrelic.common.naming.name" .) "suffix" "admission") }}
```

Comment:
🤷

Reply:
Well, I think the alternative of doing both includes everywhere is way worse... At least now it is harder to make a mistake and write the wrong suffix...

Reply:
totally agree, I was just complaining on how complex it is to pass parameters to helm helpers in general.

Reply:
A NIT could be creating other functions in the common library called `fullnameWithSuffix` that behind call `truncateToDNSWithSuffix` to make them a bit easy and don't need another helper.

Reply:
Shouldn't those functions be calling `newrelic.common.naming.fullnam` instead of `newrelic.common.naming.name`=

Reply:
We have both (the `name` one and the `fullname` one):

```
{{- define "nri-metadata-injection.name.admission" -}}
{{ include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" (include "newrelic.common.naming.name" .) "suffix" "admission") }}
{{- end -}}

{{- define  -}}
{{ include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "admission") }}
{{- end -}}
```

About having those helpers in the common.library: I'm divided, although we wouldn't need to create the helpers in each repo, we would need to call the common helper like this (repeating the suffix value in each call).

```
{{ include "newrelic.common.naming.fullnameWithSuffix" (dict "suffix" "admission" "context" .) }}
```

With specific helpers like `"nri-metadata-injection.fullname.admission"` the include is a bit easier to read and the suffix value is guaranteed as long as you use the specific helper (it can be seen as declaring some kind of a constant for the suffix value).

Reply:
It is perfect the way it is.